launch_settings:
  override_config_selection_for_this_file: False
  quit_all_if_single_node_quits: True
  verbose: False
  save_data_to_folder: True
  top_data_folder: /mnt/data_drive/video_test_data/
  prompt_notes: False
  prompt_custom_experiment_name: False
  use_date_sub_folder: True
  nodes_to_launch:
#    - package: flir_camera_driver
#      executable: publish_camera
#      name: ball_camera
    - package: video_io
      executable: video_saver
      name: video_saver
    - package: video_io
      executable: video_player
      name: video_player
#    - package: video_compressor
#      executable: video_compressor
#      name: video_compressor

ball_camera:
  ros__parameters:
    image_topic: ball_camera/image
    output_filename: .
    publish_color: false
    publish_grayscale: true
    stream_to_disk: true
    codec: h264
    burn_timestamp: true
    camera_settings:
      AcquisitionFrameRate: 60
      AcquisitionFrameRateAuto: Off
      AcquisitionFrameRateEnabled: true
      BinningVertical: 1
      GainAuto: Continuous
      LineInverter: true
      LineMode: Output
      LineSelector: Line3
      LineSource: ExposureActive
      TriggerMode: Off
    camera_chunkdata:
      FrameCounter:
        ChunkEnable: true
        ChunkModeActive: true
      Timestamp:
        ChunkEnable: true
        ChunkModeActive: true

video_player:
  ros__parameters:
    publish_frequency_double: 30.0
    loop_play: true
    image_topic: video_player/image
    publish_as_color: false
    add_timestamp: true
    verbose_logging: True
    filename: /home/maimon/maimon_ws/src/video_io/videos/fictrac_bee.mp4
    qos_image_publish_reliable: True

video_saver:
  ros__parameters:
    image_topic: video_player/image
    output_fps_double: 60.0
    quit_after_s_seconds: 30
    burn_timestamp: True
    output_filename: .
    verbose: True
    codec: nvenc_h264 # choose from [mjpeg, h264, h265, raw, nvenc_hevc]

video_compressor:
  ros__parameters:
    delete_uncompressed: False
    codec: hevc_nvenc
    crf: 23
